netty/src/main/java/io/grpc/netty/NettyWritableBuffer.java

```diff
@@ -36,10 +43,36 @@ public void write(byte[] src, int srcIndex, int length) {
   }
 
   @Override
-  public void write(byte b) {
+  public void write(int b) {
     bytebuf.writeByte(b);
   }
 
+  @Override
+  public int write(InputStream stream) throws IOException {
+    if (!bytebuf.hasArray()) {
+      if (stream instanceof BufferDrainable && bytebuf.nioBufferCount() == 1) {
+        ByteBuffer bb = bytebuf.internalNioBuffer(bytebuf.writerIndex(), bytebuf.writableBytes());
+        int positionBefore = bb.position();
+        ((BufferDrainable) stream).drainTo(bb);
+        int written = bb.position() - positionBefore;
+        bytebuf.writerIndex(bytebuf.writerIndex() + written);
+        return written;
+        // Could potentially also include composite here if stream is known length
+        // and less than first ByteBuf size. But shouldn't encounter those since
+        // output buffers come straight from the allocator
+      }
+      if (stream instanceof Drainable) {
+        return ((Drainable) stream).drainTo(this);
+      }
+    }
+    int writable = bytebuf.writableBytes();
+    int written = bytebuf.writeBytes(stream, writable);
+    if (written == writable && stream.available() != 0) {
+      throw new BufferOverflowException();
```

Comment:
Do we really need to check writable capacity at this level? Since the WritableBuffer is a wrapper backed by some real buffer and all the operations are delegated to the underlaying buffer, we should just let the underlaying buffer handle the case and such an exception should be thrown from the underneath.